src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,376 @@
+        
+        String member = "<div class='members-section'>"
+                          + "<div class='definedBy'>Defined By</div>"
+                          + "<h3 class='members-title icon-" + memberSection + "'>" + memberSectionTitle + "</h3>"
+                          + "<div class='subsection'>"
+                              + renderMember(cls, memberSection)
+                          + "</div>"
+                      + "</div>";
+        
+        return member;
+    }
+    
+    private String renderMember (SourceJSObject cls, String memberType) {
+        
+        String memberHTML = "";
+        Boolean first = true;
+        
+        for (MemberObject member : cls.members) {
+            
+            if (member.tagname.equals(memberType)) {
+
+                String firstChild = "";
+                if (first) {
+                    firstChild = "first-child";
+                    first = false;
+                }
+                
+                String inherited;
+                if (member.owner.equals(cls.name))
+                  inherited = "not-inherited";
+                else
+                  inherited = "inherited";
+                
+                String doc = "";
+                if(member.comment != null)
+                    doc = member.comment;
+                
+                String shortdoc = stripHtmlTags(doc);
+                if (shortdoc.length() > 100)
+                    shortdoc = shortdoc.substring(0, 100) + " ..."; 
+                
+                if(shortdoc.length() == 0) {
+                    shortdoc = "&nbsp;";
+                    doc      = "&nbsp;";
+                }
+                
+                if (member.meta.internal != null) 
+                    doc = renderInternal(member.meta.internal) + "</br>" + doc;
+                
+                if (member.meta.deprecated != null)
+                    doc = renderDeprecated(member.tagname, member.meta.deprecated) + "</br>" + doc;
+               
+                String sig = "";
+                String returnT = "";
+                
+                if (member.tagname.equals("method")) {
+                    if (member.parameters.size() > 0) {
+                        doc += "<br>"
+                            + "<h3 class=\"pa\">Parameters</h3>"
+                            + "<ul>";
+                    }
+                    
+                    sig = "(";
+
+                    for (ParameterObject parameter : member.parameters) {
+                        
+                        //Are we a known class being passed in? if so render a link to class
+                        String datatype;
+                        if(classes.get(parameter.datatype) != null)
+                          datatype = renderLink(parameter.datatype);
+                        else 
+                          datatype = parameter.datatype;
+                        
+                        if (sig.equals("("))

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "(" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)